components/LessonsChallenges.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import _ from 'lodash'
+import '../scss/lessonsChallenges.scss'
+
+type ChallengeStatusProps = {
+  challengesData: Challenge[]
+}
+
+type LessonChallengeProps = {
+  lessons: LessonChallenge[]
+}
+
+type LessonChallenge = {
+  order: number
+  title: string
+  totalChallenges: number
+  correctAnswers: number
+  challenges: Challenge[]
+}
+
+type Challenge = {
+  challengeNumber: number
+  challengeStatus: string
+}
+
+export const ChallengeStatus: React.FC<ChallengeStatusProps> = props => {
+  const displayChallengeStatus = props.challengesData.map(eachChallenge => {
+    let challengeStatus = 'default_challenge_status'
+    if (eachChallenge.challengeStatus === 'passed') {
+      challengeStatus = 'passed_challenge_status'
+    }
+    if (eachChallenge.challengeStatus === 'wrong answer') {
+      challengeStatus = 'unapproved_challenge_status'
+    }
+    if (eachChallenge.challengeStatus === 'pending') {
+      challengeStatus = 'pending_challenge_status'
+    }
+    return (
+      <span
+        key={_.uniqueId()}
```

Comment:
think @hwong0305 mentioned we should use something else for key

Reply:
> think @hwong0305 mentioned we should use something else for key

thats how I did it with my last PR, not sure about what he recommended. Let me know :)

Reply:
I don't see a reason why we need to use `uniqueId`. React recommends that we don't use the `index` from map for our key. But we can use either the `id` or `order` from `Challenge`.

Reply:
using `id` or `order` is like the same thing as using `index` no? using `uniqueId` forces this to rerender. @Wolfy64 brought this up last time.